test/core/utils/respondable.js

```diff
@@ -1,515 +1,561 @@
+      var errorMessage = 'Something went wrong';
+      frameSubscribe('greeting', function() {
+        throw new frameWin.TypeError(errorMessage);
       });
 
-      setTimeout(function() {
-        done();
-      }, 100);
+      respondable(
+        frameWin,
+        'greeting',
+        '',
+        true,
+        captureError(function(result) {
+          assert.instanceOf(result, TypeError);
+          assert.equal(result.message.split(/\n/)[0], errorMessage);
+          done();
+        }, done)
+      );
     });
 
-    it('should throw if an error message was send', function(done) {
-      eventData = {
-        _respondable: true,
-        _source: 'axeAPI.' + axe.version,
-        error: {
-          name: 'ReferenceError',
-          message: 'The exhaust port is open!',
-          trail: '... boom'
-        },
-        _axeuuid: 'otherAxe'
-      };
+    it('receives errors responded by the subscriber', function(done) {
+      var errorMessage = 'Something went wrong';
+      frameSubscribe('greeting', function(data, keepalive, respond) {
+        respond(new frameWin.TypeError(errorMessage));
+      });
 
-      axe.utils.respondable(win, 'Death star', null, true, function(data) {
-        assert.instanceOf(data, ReferenceError);
-        assert.equal(data.message, 'The exhaust port is open!');
+      respondable(frameWin, 'greeting', '', true, function(result) {
+        assert.instanceOf(result, TypeError);
+        assert.equal(result.message.split(/\n/)[0], errorMessage);
         done();
       });
     });
 
-    it('should create an Error if an invalid error type is passed', function(done) {
-      window.evil = function() {};
-
-      eventData = {
-        _respondable: true,
-        _source: 'axeAPI.' + axe.version,
-        error: {
-          name: 'evil',
-          message: 'The exhaust port is open!',
-          trail: '... boom'
-        },
-        _axeuuid: 'otherAxe'
-      };
+    it('can pass messages back to the subscriber (without triggering the subscriber)', function(done) {
+      frameSubscribe(
+        'greeting',
+        captureError(function(message, _, respond) {
+          assert.equal(message, '1');
+          respond('2', true, function(message) {
+            assert.equal(message, '3');
+            done();
+          });
+        }, done)
+      );
+
+      respondable(
+        frameWin,
+        'greeting',
+        '1',
+        false,
+        captureError(function(message, _, respond) {
+          assert.equal(message, '2');
+          respond('3');
+        }, done)
+      );
+    });
 
-      axe.utils.respondable(win, 'Death star', null, true, function(data) {
-        assert.instanceOf(data, Error);
-        assert.equal(data.message, 'The exhaust port is open!');
-        window.evil = undefined;
-        done();
+    it('it errors if multiple callbacks are registered', function(done) {
+      frameSubscribe('greeting', function(message, _, respond) {
```

Comment:
The back and forth nature of these tests make them confusing to follow. You have to really understand ordering implied by the callback structure. It would be more helpful if the functions were separated from the rest of the code and named to imply order. Something like:

```
function firstResponse() { respond('2a', ... }
function secondResponse() { assert.equal(message, '2a'); }
function thirdResponse() { assert.equal(message, '3a'); }
```